cloudconfig/userdatacfg_unix.go

```diff
@@ -321,18 +321,22 @@ func (w *unixConfigure) ConfigureJuju() error {
 		// /etc/juju-proxy.conf file will be empty.
 		`[ -e /etc/profile.d/juju-proxy.sh ] || ` +
 			`printf '\n# Added by juju\n[ -f "/etc/juju-proxy.conf" ] && . "/etc/juju-proxy.conf"\n' >> /etc/profile.d/juju-proxy.sh`)
-	if w.icfg.LegacyProxySettings.HasProxySet() {
-		exportedProxyEnv := w.icfg.LegacyProxySettings.AsScriptEnvironment()
-		w.conf.AddScripts(strings.Split(exportedProxyEnv, "\n")...)
-		w.conf.AddScripts(
-			fmt.Sprintf(
-				`(printf '%%s\n' %s > /etc/juju-proxy.conf && chmod 0644 /etc/juju-proxy.conf)`,
-				shquote(w.icfg.LegacyProxySettings.AsScriptEnvironment())))
-
-		// Write out systemd proxy settings
-		w.conf.AddScripts(fmt.Sprintf(`printf '%%s\n' %[1]s > /etc/juju-proxy-systemd.conf`,
-			shquote(w.icfg.LegacyProxySettings.AsSystemdDefaultEnv())))
+	var proxySettings proxy.Settings
+	if w.icfg.JujuProxySettings.HasProxySet() {
+		proxySettings = w.icfg.JujuProxySettings
+	} else if w.icfg.LegacyProxySettings.HasProxySet() {
+		proxySettings = w.icfg.LegacyProxySettings
 	}
+	exportedProxyEnv := proxySettings.AsScriptEnvironment()
+	w.conf.AddScripts(strings.Split(exportedProxyEnv, "\n")...)
```

Comment:
We only want to do this for legacy settings I think, right?

Reply:
If I am not mistaken, this affects the cloud-init shell only (`runcmd`) on first boot.

```
		exportedProxyEnv := proxySettings.AsScriptEnvironment()
		w.conf.AddScripts(strings.Split(exportedProxyEnv, "\n")...)
```

About this one I am not sure since there are potential `preruncmd` and `postruncmd` commands added by a user via model-config.

If we follow the same principle here, I think we could:

* expose JUJU_CHARM_{HTTP, HTTPS, NO}_PROXY in the cloud-init shell environment and document that those should be used explicitly;
* tell users that /etc/juju-proxy.conf can be sourced in any custom `postruncmd`s.
    * wouldn't work on Windows since we don't write a file there.